pkg/server/server.go

```diff
@@ -910,7 +910,7 @@ func (s *Server) Start(ctx context.Context) error {
 	s.stopper.AddCloser(stop.CloserFn(gwCancel))
 
 	// Setup HTTP<->gRPC handlers.
-	conn, err := s.rpcContext.GRPCDial(s.cfg.Addr)
+	conn, err := s.rpcContext.GRPCDial(s.cfg.AdvertiseAddr)
```

Comment:
are you sure that's right? Depending on Addr vs AdvertiseAddr and the network topology, this may force local connections to go through an external interface, even a firewall.